### PR TITLE
fix(deadline): Fix cyclic stack dependency when using UBL

### LINKED ISSUE #245: Cyclic dependency when using UsageBasedLicensing
Given a tiered architecture like in the RFDK examples where the UsageBasedLicensing is in a prerequisite tier of the WorkerInstanceFleet, attempts to use UBL's [grantPortAccess() method](https://docs.aws.amazon.com/rfdk/api/latest/docs/aws-rfdk.deadline.UsageBasedLicensing.html#grant-wbr-port-wbr-accessworkerfleet-licenses) result in a cyclic dependency between the Worker's stack & the UBL stack. This cyclic dependency is caused by the security group ingress rule that must be created on the UBL's security group.

### Reproduction Steps

Add a `grantPortAccess()` call to an RFDK's example app.

### Error Log

Cyclic dependency.

### Environment

  - **CDK CLI Version      :** 1.72.0
  - **CDK Framework Version:** 1.72.0
  - **RFDK Version:** 0.20.0
  - **Deadline Version:** 10.1.11
  - **Node.js Version:** any
  - **OS               :** any
  - **Language (Version):** all

### Other

I will fix this.

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by ddneilson**
Fixes: https://github.com/aws/aws-rfdk/issues/245

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-rfdk/lib/deadline/lib/usage-based-licensing.ts [hunk 1]
```diff
@@ -594,10 +593,12 @@ export class UsageBasedLicensing extends Construct implements IGrantable {
   public grantPortAccess(workerFleet: IWorkerFleet, licenses: UsageBasedLicense[]) {
     licenses.forEach(license => {
       license.ports.forEach(port => {
-        this.connections.allowFrom(workerFleet, port);
+        // this.connections.allowFrom(workerFleet, port);
```
Comment (jusiskin) [inline packages/aws-rfdk/lib/deadline/lib/usage-based-licensing.ts]: nit: we can remove this commented-out old/broken implementation. same goes for below.

**Review by jusiskin — DISMISSED** [review]
Thanks for fixing this! Just a minor clean-up and looks good.

File: packages/aws-rfdk/lib/deadline/lib/usage-based-licensing.ts (hunk 1 above)
Comment (ddneilson) [reply packages/aws-rfdk/lib/deadline/lib/usage-based-licensing.ts]: Yup, yup. That's why I marked the PR as a Draft. ;-) 
I also still have to validate the changes to the example apps.

File: packages/aws-rfdk/lib/deadline/test/usage-based-licensing.test.ts [hunk 2]
```diff
@@ -495,34 +603,38 @@ describe('UsageBasedLicensing', () => {
...
       });
 
+      // WHEN
       ubl.grantPortAccess(workerFleet, [license]);
 
       // THEN
       ports.forEach( port => {
-        const ublAsgSecurityGroup = ubl.asg.connections.securityGroups[0].node.defaultChild;
-        const ublAsgSecurityGroupLogicalId = isolatedStack.getLogicalId(ublAsgSecurityGroup as CfnElement);
+        // const ublAsgSecurityGroup = ubl.asg.connections.securityGroups[0].node.defaultChild;
```
Comment (ddneilson) [inline packages/aws-rfdk/lib/deadline/test/usage-based-licensing.test.ts]: Note to self. Remove these commented-out lines.

File: packages/aws-rfdk/lib/deadline/test/usage-based-licensing.test.ts [hunk 3]
```diff
@@ -545,7 +657,15 @@ describe('UsageBasedLicensing', () => {
...
-      const ublAsgSg = ubl.asg.connections.securityGroups[0].node.defaultChild as CfnElement;
+      // WHEN
+      new UsageBasedLicensing(stack, 'UBL', {
+        certificateSecret,
+        images,
+        licenses,
+        renderQueue,
+        vpc,
+      });
+      // const ublAsgSg = ubl.asg.connections.securityGroups[0].node.defaultChild as CfnElement;
```
Comment (ddneilson) [inline packages/aws-rfdk/lib/deadline/test/usage-based-licensing.test.ts]: Note to self. Remove commented-out line.

File: packages/aws-rfdk/lib/deadline/lib/usage-based-licensing.ts (hunk 1 above)
Comment (jusiskin) [reply packages/aws-rfdk/lib/deadline/lib/usage-based-licensing.ts]: oops. missed the draft marker - sorry

**Review by horsmand — APPROVED** [review]

**Review by yashda — APPROVED** [review]
Changes looks good to me. Thanks Daniel!

**Pull request merged**